### ISSUE TITLE
Restrângerea rezultatelor în cazul a mai mult de X intrări

### ISSUE THREAD

**Issue by LadislauStrifler** [dexonline/dexonline#811]
În cazul unor căutări de cuvinte de legătură (prepoziții, conjuncții ș.a.) ex.: [de](https://dexonline.ro/definitie/de), lista rezultatelor oferite este prea lungă. Propun restrângerea rezultatelor la lista lexemelor simple, urmând ca din lista celor compuse să se afișeze primele 2-3, accesul la lista completă să fie astfel condiționată de apăsarea unui buton „arată toate intrările” pe modelul definițiilor.
![image](https://user-images.githubusercontent.com/16460700/50354110-5a270d80-0542-11e9-9167-52ea506f848e.png)

**Comment by MateiGall** [comment]
👍
*Matei GALL*
* <http://dexonline.ro>*
*http://dexonline.ro <http://dexonline.ro>*

**Comment by CatalinFrancu** [comment]
Lucrurile se complică un pic din cauză că acum vor exista patru stări:

1. Maxim 10 intrări, maximum 20 de definiții din aceste intrări.
2. Maxim 10 intrări, toate definițiile din aceste intrări.
3. Toate intrările, maximum 20 de definiții din aceste intrări (pot fi altele decît în starea 1).
4. Toate intrările, toate definițiile.

Nici forma URL-ului nu e foarte clară. În prezent afișarea tuturor definițiilor este marcată prin `/expandat` în URL. Ar trebui să avem două astfel de particule.

Ce-ar fi să existe **un singur** buton, _arată toate definițiile și intrările_? Deci am avea doar stările 1 și 4.

**Comment by CatalinFrancu** [comment]
Aș zice că cele două sînt aproape una și aceeași problemă. Cînd ai mai multe intrări la rezultate, mare parte dintre ele sînt compuse (la fel și la iarbă, mare, stea etc.). Deci orice soluție găsim la problema generală (sînt prea multe intrări), cred că va rezolva și problema intrărilor compuse.

Aș mai preciza că lista cu definiții începe cu cele pentru intrarea simplă, ceea ce cred că e bine. Deci deranjul apare doar fiindcă lista de intrări e lungă și împinge toate rezultatele în jos.

**Comment by LadislauStrifler** [comment]
Nu, n-am propus să existe același tratament pentru compuse. Butonul va avea descrierea „afișează 500 de compuse”. La acționare lui toate intrările compuse vor fi afișate. 
Într-un moment de nebunie am putea implementa paginare.

**Comment by CatalinFrancu** [comment]
Păi nu, eu am propus. :-)

Sau am putea păstra butonul ca pînă acum, să controleze doar trecerea de la 20 de definiții la toate, iar pentru intrări să avem un link de genul _+ 150 de alte intrări ▼_, care la click expandează un div.

**Comment by LadislauStrifler** [comment]
Dacă cele 20 de definiții, din faza inițială, sunt toate asociate cu intrările „simple” atunci parcă ar fi în regulă cu expandarea div-ului.
Eu mă gândeam la un tratament complet separat pentru compuse. Dar un tab separat cum ar arăta?

**Comment by CatalinFrancu** [comment]
Parcă nu-mi place ideea de a separa definițiile în mai multe taburi. În prezent taburile conțin elemente radical diferite (definiții, paradigme, arbori).

Într-adevăr, primele 20 de definiții nu sînt în mod garantat asociate cu intrarea simplă. În principiu sînt, căci primează definițiile pentru cuvîntul căutat („iarbă” înaintea compuselor). Dar pot fi sub 20 (vezi fonogramic, fantastic etc.)
